bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5375,28 +5382,22 @@ void ReplicaImp::onExecutionFinish() {
     const auto &seq_num_to_create_dbcheckpoint = DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot();
     if (seq_num_to_create_dbcheckpoint.has_value()) {
       if (seq_num_to_create_dbcheckpoint.value() > (lastExecutedSeqNum + activeExecutions_)) {
-        DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(lastExecutedSeqNum + activeExecutions_ + 1,
-                                                                          true);  // noop=true
-      } else {
-        onSeqNumIsStableCallbacks_.add([seq_num_to_create_dbcheckpoint](SeqNum seqNum) {
-          if (seqNum == seq_num_to_create_dbcheckpoint) {
-            DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(seqNum, false);  // noop=false
-            LOG_INFO(GL,
-                     "sendInternalCreateDbCheckpointMsg with noop: false, "
-                         << KVLOG(seq_num_to_create_dbcheckpoint.value()));
-          }
-        });
-        DbCheckpointManager::instance().setNextStableSeqNumToCreateSnapshot(std::nullopt);
+        std::string cid = "replicaDbCheckpoint_" + std::to_string(lastExecutedSeqNum + activeExecutions_ + 1) + "_" +
+                          std::to_string(config_.replicaId);
+        sendInternalNoopCommand(cid);
       }
-      LOG_INFO(GL,
-               "sendInternalCreateDbCheckpointMsg: " << KVLOG(
-                   lastExecutedSeqNum, activeExecutions_, seq_num_to_create_dbcheckpoint.value()));
     }
   }
 
   if (isCurrentPrimary() && requestsQueueOfPrimary.size() > 0) tryToSendPrePrepareMsg(true);
 }
 
+void ReplicaImp::sendInternalNoopCommand(const std::string &cid) {
+  std::string strMsg;
```

Comment:
Wasn't this code already implemented for wedges?
If not, why do these noops include a new flag?

Reply:
Wedge noops are going through a different code path (for example, they are not completely ignored and create new blocks). 
I agree that eventually, to use this new NOOP flag in wedge also, but I prefer to make it in a different PR